### PR TITLE
[enzyme] Add renderProp() function

### PR DESCRIPTION
**Pull request by screendriver**
I know there was already a pull request https://github.com/DefinitelyTyped/DefinitelyTyped/pull/32963/ for that but	unfortunately the `Parameters` suggestion is only available in TypeScript 3.1 and not in 2.8. Any help how to solve this in 2.8 would be welcome ☺️ 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/airbnb/enzyme/blob/master/docs/api/ShallowWrapper/renderProp.md
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by jwbay** [comment]
> Any help how to solve this in 2.8 would be welcome

This was unclear from my comment, but I was trying to suggest just inlining that Parameters helper into Enzyme's definitions. We might need to bump the version to something newer just to get the argument tuple support, though. If we do, may as well bump all the way to 3.1 since 3.3 is out.

**Review by jwbay — CHANGES_REQUESTED** [review]

**Comment by screendriver** [comment]
@jwbay thanks for your review. I tried to bump the TypeScript version to `3.1` and add your changes but unfortunately I get the error `chai-enzyme depends on enzyme but has a lower required TypeScript version.`. Either we find a solution that works with `2.8` or upgrade `chai-enzyme` as well.

What do you think?

File: types/enzyme/index.d.ts [hunk 1]
```diff
@@ -364,6 +364,8 @@ export interface CommonWrapper<P = {}, S = {}, C = Component<P, S>> {
     length: number;
 }
 
+type Parameters<T> = T extends (...args: infer A) => any ? A : never
```
Comment (jwbay) [inline types/enzyme/index.d.ts]: This is built into TS version 3.1, so doesn't need to be declared inline _along with_ a bump to 3.1. We only need one or the other. The helper should be inlined like this only if we want to support a slightly older version of TS (2.9 or 3.0, I believe). Sorry for the confusion.

I'm fine with keeping the version at 3.1 and dropping this helper.

**Comment by jwbay** [comment]
The minimum version for `chai-enzyme` can be upgraded as well. We're well within DT policy to bump things past 2.8.

**Comment by screendriver** [comment]
@jwbay I fixed a lot more 😅 But now it should work. Could you please have a look at it?

File: types/enzyme/index.d.ts [hunk 2]
```diff
@@ -363,6 +364,8 @@ export interface CommonWrapper<P = {}, S = {}, C = Component<P, S>> {
     length: number;
 }
 
+export type Parameters<T> = T extends (...args: infer A) => any ? A : never;
```
Comment (jwbay) [inline types/enzyme/index.d.ts]: We should be able to remove this now that we're on TS version 3.1 👍

Comment (screendriver) [reply types/enzyme/index.d.ts]: Does not work :confused: You can see the output in the last Travis build.

Comment (jwbay) [reply types/enzyme/index.d.ts]: Oh, interesting. The inlined version doesn't have a function constraint for the type parameter like the built-in version does. This is actually important for our case. Let's add the inlined helper back and we'll be ready to ship. Sorry for the back and forth!

Comment (screendriver) [reply types/enzyme/index.d.ts]: No problem ☺️ I pushed a new version with the inline helper back again 😉

**Review by jwbay — APPROVED** [review]

**Comment by amcasey** [comment]
So the upgrade to TS 3.1 is still required in the final version of this change?

**Comment by jwbay** [comment]
We might be able to get away with 3.0 instead of 3.1, but no older.

**Comment by amcasey** [comment]
@jwbay Thanks for confirming!

**Pull request merged**